src/shogun/labels/BinaryLabels.cpp

```diff
@@ -148,3 +149,53 @@ CLabels* CBinaryLabels::shallow_subset_copy()
 
 	return shallow_copy_labels;
 }
+
+namespace shogun
+{
+	Some<CBinaryLabels> binary_from_binary(CBinaryLabels* orig)
+	{
+		return Some<CBinaryLabels>::from_raw(orig);
+	}
+
+	Some<CBinaryLabels> binary_from_dense(CDenseLabels* orig)
+	{
+		auto labels = orig->get_labels();
+		for (auto i : range(labels.vlen))
+		{
+			if (!CMath::fequals(labels[i], 1.0, 0.0) &&
+			    !CMath::fequals(labels[i], -1.0, 0.0))
+			{
+				SG_SERROR(
+				    "Label[%d] is %f, but must be either +1.0 or -1.0.\n", i,
+				    labels[i])
+			}
+		}
+		auto result = new CBinaryLabels(labels);
```

Comment:
afaik this copies the whole vvector itself.... do we really need to copy the data where it doesn't change at all....

Reply:
it copies indeed, which I actually didnt realise. Why does it copy? Doesnt seem to make sense to me. And agree, it should not